### PR TITLE
EasyBuild roles improvement 

### PR DESCRIPTION
**Pull request by issamsaid**
- removed all the hardcoded shell commands
- removed some unnecessary tasks
- added support for arm64 (jetson+EGX)

### CODE REVIEW

File: roles/lmod/templates/lmod-ubuntu.j2 [hunk 1]
```diff
@@ -15,10 +12,10 @@ if [ $(id -u) -ne 0 ] ; then
     export LMOD_FULL_SETTARG_SUPPORT=no
     export LMOD_COLORIZE=yes
     export LMOD_PREPEND_BLOCK=normal
-    export MODULEPATH_ROOT="{{ sm_module_root }}"
-    export MODULEPATH="{{ sm_module_path }}"
+    export MODULEPATH_ROOT="/sw/modules"
```
Comment (supertetelman) [inline roles/lmod/templates/lmod-ubuntu.j2:15]: Is there a reason we replaced this variable with a hardcoded path?

**Comment by supertetelman** [comment]
This all looks good to me.

Is there an easy way we could test this internally (or better yet, add a Jenkins test)?

File: roles/lmod/templates/lmod-ubuntu.j2 (hunk 1 above)
Comment (issamsaid) [reply roles/lmod/templates/lmod-ubuntu.j2]: good catch, it was a leftover. Fixed

File: roles/easy-build/tasks/main.yml [hunk 2]
```diff
@@ -12,98 +21,32 @@
...
+# ansible doesn't run task in a login shell, so we have to use sudo -ilc
+- name: "install latest easybuild version"
+  shell: /bin/bash -ilc '/tmp/bootstrap_eb.py {{ sm_prefix }}'
   when: not ebsw_exe.matched
 
-- name: "get EasyBuild version"
-  become: yes
-  shell: ls {{ eb_tmp_dir }}/software/EasyBuild/
+- name: "get easybuild version"
+  shell: ls {{ sm_prefix }}/software/EasyBuild/
```
Comment (dholt) [inline roles/easy-build/tasks/main.yml:47]: Will this work if there's more than one version of EB installed?

File: roles/easy-build-packages/tasks/main.yml [hunk 3]
```diff
@@ -39,11 +39,6 @@
   with_items: "{{ eb_meta.files }}"
 
 - name: "install the selected easybuild modules"
-  shell: |  
-    . /etc/profile.d/z00_lmod.sh && 
-    . /etc/profile.d/z01_eb.sh   && 
-    eb --parallel=16 "{{ item }}" --robot
+  shell: /bin/bash -ilc 'module load EasyBuild && eb --prefix={{ sm_prefix }} --parallel=16 "{{ item }}" --robot'
```
Comment (dholt) [inline roles/easy-build-packages/tasks/main.yml:42]: Is `sm_prefix` defined somewhere? It should have a default set in the role

File: roles/lmod/tasks/main.yml [hunk 4]
```diff
@@ -12,66 +25,98 @@
...
+    path: "{{ item.path }}"
+    regexp: 'x86_64-linux-gnu'
+    replace: 'aarch64-linux-gnu'
+  with_items: "{{ lua_config.files }}"
+  when: ansible_os_family == "Debian" and ansible_architecture == "aarch64"
+  tags:
+    - bugfix
+
 - name: "mkdir software path"
   file: path={{ sm_software_path }} state=directory
```
Comment (dholt) [inline roles/lmod/tasks/main.yml:71]: Is `sm_software_path` defined somewhere? It should have a default set in the role

File: roles/lmod/templates/lmod-redhat.j2 [hunk 5]
```diff
@@ -31,7 +27,7 @@ if ( ! $?MODULEPATH_ROOT ) then
     setenv MODULEPATH_ROOT      "{{ sm_module_root }}"
     # setenv MODULEPATH           `/usr/share/lmod/lmod/libexec/addto --append MODULEPATH $MODULEPATH_ROOT/$LMOD_sys $MODULEPATH_ROOT/Core`
     # setenv MODULEPATH           `/usr/share/lmod/lmod/libexec/addto --append MODULEPATH /usr/share/lmod/lmod/modulefiles/Core`
-    setenv MODULEPATH           "{{ eb_modulepath }}"
+    setenv MODULEPATH           "{{ sm_module_path }}"
```
Comment (dholt) [inline roles/lmod/templates/lmod-redhat.j2:30]: Is Is `sm_module_path` defined somewhere? It should have a default set in the role

File: roles/lmod/templates/lmod-redhat.j2 [hunk 6]
```diff
@@ -31,7 +27,7 @@ if ( ! $?MODULEPATH_ROOT ) then
     setenv MODULEPATH_ROOT      "{{ sm_module_root }}"
```
Comment (dholt) [inline roles/lmod/templates/lmod-redhat.j2:27]: Is `sm_module_root` defined somewhere? It should have a default set in the role

Comment (issamsaid) [reply roles/lmod/templates/lmod-redhat.j2:27]: working on it.

File: roles/lmod/tasks/main.yml (hunk 4 above)
Comment (supertetelman) [reply roles/lmod/tasks/main.yml:89]: https://github.com/NVIDIA/deepops/pull/543/commits/af22cdf54b5490ac37f2a040e71b8db90563a640

File: config.example/group_vars/all.yml [hunk 7]
```diff
@@ -221,6 +221,17 @@ maas_single_node_install: true
...
 
+################################################################################
+# SOFTWARE MODULES (SM)                                                        #
+################################################################################
+sm_prefix: /sw                                 # has to be in a shared location
+sm_module_root: "{{ sm_prefix }}/modules"
+sm_module_path: "{{ sm_module_root }}/all"
+sm_software_path: "{{ sm_prefix }}/software"
+sm_files_path: "{{ sm_prefix }}/easyconfigs"
+sm_files_url: "https://github.com/DeepOps/easyconfigs.git"
```
Comment (supertetelman) [inline config.example/group_vars/all.yml:232]: What is this? @dholt, should this stuff be moved over to a /NVIDIA git repo?

File: playbooks/nvidia-driver.yml [hunk 8]
```diff
@@ -9,6 +9,11 @@
       stat:
         path: /etc/dgx-release
       register: is_dgx
+    
+    - name: Check for Jetson packages
+      stat:
+        path: /etc/nv_tegra_release
+      register: is_jetson
```
Comment (supertetelman) [inline playbooks/nvidia-driver.yml:16]: This probably shouldn't be in the EB PR, but it seems okay.

File: roles/easy-build-packages/defaults/main.yml [hunk 9]
```diff
@@ -1 +1,7 @@
-sm_files_path: "{{ sm_prefix }}/easybuild_files"
+sm_prefix: /sw
+sm_files_url: "https://github.com/DeepOps/easyconfigs.git"
```
Comment (supertetelman) [inline roles/easy-build-packages/defaults/main.yml:2]: @dholt , same here. Is this the right GitHub?

**Review by supertetelman — APPROVED** [review]

**Comment by michael-balint** [comment]
Hey @issamsaid - we recently did a reorg of the DeepOps directory structure. To fix the merge conflicts above, you'll want to move `playbooks/nvidia-drivery.yml` --> `playbooks/nvidia-software/nvidia-driver.yml` (and remove kubespray since it has moved to `submodules/kubespray` and is tracked as a submodule). Keep in mind that also any references to scripts/playbooks need to reflect the recent changes to directory structure. Let me know if you have any questions.

**Comment by supertetelman** [comment]
Closing this PR so that I can make a new one that addresses the re-org.

**Pull request closed without merging**